### PR TITLE
Reenabled Conjur HTTP connector

### PR DESCRIPTION
**Pull request by sgnn7**
This commit converts the old-style Conjur handler/listener to the new
connector API.

#### What ticket does this PR close?
Connected to  #859 

#### Where should the reviewer start?
[Jenkins Build](https://jenkins.conjur.net/blue/organizations/jenkins/cyberark--secretless-broker/detail/859-conjur-http-connector/1/pipeline)

#### What is the status of the manual tests?
Have you run the following manual tests to verify existing functionality continues to function as expected?
- [ ] Manually tested [Keychain provider](https://github.com/cyberark/secretless-broker/tree/master/test/manual/keychain_provider)
- [ ] Manually run the [K8s demo](https://github.com/cyberark/secretless-broker/tree/master/demos/k8s-demo) (which requires changing the image the demo uses to the local build of Secretless)
- [ ] Manually run [Kubernetes-Conjur demo](https://github.com/conjurdemos/kubernetes-conjur-demo) with a local Secretless Broker image build of your branch

If this feature does not have any/sufficent automated tests, have you created/updated a folder in `test/manual` that includes:
- [ ] An updated README with instructions on how to manually test this feature
- [ ] Utility `start` and `stop` scripts to spin up and tear down the test environments
- [ ] A `test` script to run some basic manual tests (optional; if does not exist, the README should have detailed instructions)

### CODE REVIEW

File: internal/proxyservice/http/conjur/connector.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	"encoding/base64"
+	"fmt"
+	gohttp "net/http"
+	"strconv"
+
+	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+)
+
+// Connector injects an HTTP request with Cponjur's authorization header(s).
```
Comment (doodlesbykumbi) [inline internal/proxyservice/http/conjur/connector.go]: `s/Cponjur's/Conjur's`/

File: internal/proxyservice/http/conjur/connector.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+type Connector struct {
+	logger log.Logger
+}
+
+// Connect is the function that implements the http.Connector func
+// signature. It has access to the client http.Request and the credentials (as a
+// map), and is expected to decorate the request with Authorization headers.
+//
+// Connect uses the "accessKeyId", "secretAccessKey" and optional "accessToken"
```
Comment (doodlesbykumbi) [inline internal/proxyservice/http/conjur/connector.go]: Don't forget to change from AWS to Conjur appropriate comment

File: internal/proxyservice/http/conjur/connector.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	var ok bool
+
+	accessToken, ok := credentialsByID["accessToken"]
+	if !ok {
+		return fmt.Errorf("Conjur credential 'accessToken' is not available")
+	}
+
+	forceSSLStr, ok := credentialsByID["forceSSL"]
+	forceSSL, err := strconv.ParseBool(string(forceSSLStr))
+	if ok && err == nil && forceSSL {
```
Comment (doodlesbykumbi) [inline internal/proxyservice/http/conjur/connector.go]: Perhaps work for later but this should probably become a generic http proxy service option

File: internal/proxyservice/http/conjur/connector.go (hunk 2 above)
Comment (sgnn7) [reply internal/proxyservice/http/conjur/connector.go]: Oops!

File: internal/proxyservice/http/conjur/connector.go (hunk 3 above)
Comment (sgnn7) [reply internal/proxyservice/http/conjur/connector.go]: Yup - probably for later

**Comment by sgnn7** [comment]
@doodlesbykumbi Updated the branch with fixes. CR ready again.

File: internal/proxyservice/http/conjur/connector.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+// header.
+func (c *Connector) Connect(
+	req *gohttp.Request,
+	credentialsByID connector.CredentialValuesByID,
+) error {
+	var ok bool
+
+	accessToken, ok := credentialsByID["accessToken"]
+	if !ok {
+		return fmt.Errorf("Conjur credential 'accessToken' is not available")
```
Comment (doodlesbykumbi) [inline internal/proxyservice/http/conjur/connector.go]: Go lint nit: errors start with lowercase

**Pull request merged**